grid-cli/cmd/deploy_kubernetes.go

```diff
@@ -200,17 +200,20 @@ var deployKubernetesCmd = &cobra.Command{
 				disks,
 				nil,
 				rootfss,
-				uint64(len(workers)),
+				uint64(len(workers)-len(workersNode)),
 			)
 			if err != nil {
 				log.Fatal().Err(err).Send()
 			}
-			for i, node := range workersNodes {
-				workers[i].Node = uint32(node.NodeID)
+			for i := 0; i < len(workersNode); i++ {
+				workers[i].Node = uint32(workersNode[i])
+			}
+			for i, j := len(workersNode), 0; i < len(workers); i++ {
+				workers[i].Node = uint32(workersNodes[j].NodeID)
 			}
 		} else {
 			for i := 0; i < workerNumber; i++ {
-				workers[i].Node = workersNode
+				workers[i].Node = uint32(workersNode[i])
```

Comment:
you can append the filtered nodes to the workers nodes then use a one loop over all workers to assign the node id, you won't need else then